src/gc/gc.d

```diff
@@ -3132,19 +3134,17 @@ struct SmallObjectPool
 
                 rt_finalizeFromGC(q, size, attr);
 
-                toClear |= GCBits.BITS_1 << clearIndex;
+                freeBits = true;
+                toFree.set(i);
 
                 debug(COLLECT_PRINTF) printf("\tcollecting %p\n", p);
                 //log_free(sentinel_add(p));
 
                 debug (MEMSTOMP) memset(p, 0xF3, size);
-                freebits.set(biti);
```

Comment:
You must not hold references to objects created from a shared library once it is unloaded. Not freeing the objects isn't safer, it just hides bugs.